docs/certificate support proposal.md

```diff
@@ -0,0 +1,229 @@
+      - a scope corresponding to a V1 application (for instance <GUID>/user_impersonation, where  <GUID> is the
+        clientId of a V1 application, created in the https://portal.azure.com portal.
+    */
+    "TodoListScope": "api://a4c2469b-cf84-4145-8f5f-cb7bacf814bc/access_as_user",
+    "TodoListBaseAddress": "https://localhost:44351"
+
+  },
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft": "Warning",
+      "Microsoft.Hosting.Lifetime": "Information"
+    }
+  },
+  "AllowedHosts": "*"
+}
+
+```
+
+## Todo:
+- provide feedback on this proposal. Would we rather want to have a service to get certificates?
+- Come-up with MicrosoftIdentityOptions to read this
+- Read the certificate from its description.
+- Apply the certificate to ConfidentialClientApplication.
+
+See also [Sample code to load certiticates](https://github.com/AzureAD/microsoft-identity-web/wiki/Spec-certificates#sample-code-to-load-certificates) - which is not necessarily what we want to do, but gives an idea on manipulate the certs?
+
+See also 
+- https://github.com/AzureAD/microsoft-identity-web/issues/72
+- https://github.com/AzureAD/microsoft-identity-web/issues/12
+
+
+## Proposal/Principles
+Initially let's reduce the problem by saying that:
+- we want to support both ClientCertificates and EncryptionCertificate collections in the configuration (in MicrosoftIdentityOptions)
+- however initially we'll only consider the **first** certificate of each collection. We would not handle the rotation of certificate yet, as this would need to be pushed to MSAL.NET (for the ClientCertificate), and to Wilson (for the EncyrptionCertificate). We just add the collection so that we don't break the public API later.
+- we would start by only supporting certificates obtained from KeyVault, as this is the use case that most customers are after. But we'd propose a design that would work with other source of certificates (local file, base64 encoded are the ones I've seen the most)
+
+Proposed sequence:
+1. Experiment with the KeyVault SDK part of the Azure SDK. This has the advantage of getting the secrets from KeyVault both locally, and when the Web app or Web API is deployed, as it leverages Managed identity. See https://docs.microsoft.com/en-us/azure/key-vault/general/tutorial-net-create-vault-azure-web-app#update-the-code (this one is for client secrets, we need to find the equivalent with certificates, for instance by looking at https://docs.microsoft.com/en-us/dotnet/api/overview/azure/security.keyvault.certificates-readme?view=azure-dotnet#retrieve-a-certificate). We could use, for the client the DefaultAzureCredentials (which also work well in the dev environment)
+
+   This means we would now leverage the following NuGet packages: **Azure.Identity** and **Azure.Security.KeyVault.Certificates**. This seems fine but we need to check if there is a negative impact.
```

Comment:
As part of the experiment, we should also understand if there is significant logic related to MSI in those assemblies. If not, perhaps we can avoid the dependencies.